### ISSUE TITLE
'atomic scan' blowing up

### ISSUE THREAD

**Issue by miabbott** [projectatomic/atomic#925]
Using git master, I observed `atomic scan` blowing up using both `registry.access.redhat.com/rhel7/openscap` or `docker.io/fedora/atomic_scan_openscap` images.  Strangely, they blow up in different ways.

Using `registry.access.redhat.com/rhel7/openscap`

```
# ./atomic --debug scan --scanner openscap registry.access.redhat.com/rhel7
Created /run/atomic/2017-03-01-22-10-14-859809
docker run -t --rm -v /etc/localtime:/etc/localtime -v /run/atomic/2017-03-01-22-10-14-859809:/scanin -v /var/lib/atomic/openscap/2017-03-01-22-10-14-859809:/scanout:rw,Z -v /etc/oscapd:/etc/oscapd:ro registry.access.redhat.com/rhel7/openscap oscapd-evaluate scan --no-standard-compliance --targets chroots-in-dir:///scanin --output /scanout
Created /run/atomic/2017-03-01-22-10-14-859809/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3
Mounted e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3 to /run/atomic/2017-03-01-22-10-14-859809/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3
Creating the output dir at /var/lib/atomic/openscap/2017-03-01-22-10-14-859809
INFO:OpenSCAP Daemon one-off evaluator 0.1.6
WARNING:Can't import the 'docker' package. Container scanning functionality will be disabled.
INFO:Creating tasks directory at '/var/lib/oscapd/tasks' because it didn't exist.
INFO:Creating results directory at '/var/lib/oscapd/results' because it didn't exist.
INFO:Creating results work in progress directory at '/var/lib/oscapd/work_in_progress' because it didn't exist.
INFO:Evaluated EvaluationSpec, exit_code=0.
INFO:Evaluated EvaluationSpec, exit_code=0.
INFO:[100.00%] Scanned target 'chroot:///scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3'
Unmounted /run/atomic/2017-03-01-22-10-14-859809/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3
u'/scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3'
Traceback (most recent call last):
  File "./atomic", line 189, in <module>
    sys.exit(_func())
  File "/root/atomic/Atomic/scan.py", line 166, in scan
    self.output_results()
  File "/root/atomic/Atomic/scan.py", line 274, in output_results
    uuid = self.mount_paths[json_results['UUID'].rstrip('/')] if len(self.args.rootfs) == 0 \
KeyError: u'/scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3'
```

Using `docker.io/fedora/atomic_scan_openscap`

```
 ./atomic --debug scan --scanner atomic_scan_openscap registry.access.redhat.com/rhel7
Created /run/atomic/2017-03-02-14-24-44-627828
docker run -t --rm -v /etc/localtime:/etc/localtime -v /run/atomic/2017-03-02-14-24-44-627828:/scanin -v /var/lib/atomic/atomic_scan_openscap/2017-03-02-14-24-44-627828:/scanout:rw,Z -v /etc/oscapd:/etc/oscapd:r
o docker.io/fedora/atomic_scan_openscap oscapd-evaluate scan --no-standard-compliance --targets chroots-in-dir:///scanin --output /scanout
Created /run/atomic/2017-03-02-14-24-44-627828/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3
Mounted e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3 to /run/atomic/2017-03-02-14-24-44-627828/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3
Creating the output dir at /var/lib/atomic/atomic_scan_openscap/2017-03-02-14-24-44-627828
INFO:OpenSCAP Daemon one-off evaluator 0.1.6
WARNING:Can't import the 'docker' package. Container scanning functionality will be disabled.
INFO:Creating tasks directory at '/var/lib/oscapd/tasks' because it didn't exist.
INFO:Creating results directory at '/var/lib/oscapd/results' because it didn't exist.
INFO:Creating results work in progress directory at '/var/lib/oscapd/work_in_progress' because it didn't exist.
ERROR:EvaluationSpec failed to evaluate, oscap returned 1 as exit code, it may not be possible to get ARF/OVAL results or generate reports for this result!
ERROR:Failed to detect CPEs of target 'chroot:///scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3'. Assuming no CPEs...
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 493, in evaluate
    encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/oscapd/work_in_progress/mr3cgdf1/results.xml'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/oscapd-evaluate", line 129, in scan_worker
    detect_CPEs_of_target(target, config)
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 529, in detect_CPEs_of_target
    results, stdout, stderr, exit_code = es.evaluate(config)
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 499, in evaluate
    (stdout, stderr, e)
RuntimeError: Failed to read results.xml of EvaluationSpec evaluation.
stdout:


stderr:
OpenSCAP Error: Probe at sd=32 (system_info) reported an error: Invalid type, value or format [oval_probe_ext.c:393]
Unable to receive a message from probe [oval_probe_ext.c:579]
Failed to create a new agent session. [oval_session.c:320]


exception: [Errno 2] No such file or directory: '/var/lib/oscapd/work_in_progress/mr3cgdf1/results.xml'
ERROR:EvaluationSpec failed to evaluate, oscap returned 1 as exit code, it may not be possible to get ARF/OVAL results or generate reports for this result!
ERROR:Failed to scan target 'chroot:///scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3' for vulnerabilities.
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 493, in evaluate
    encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/oscapd/work_in_progress/x8zu0__d/results.xml'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/oscapd-evaluate", line 143, in scan_worker
    es.evaluate(config)
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 473, in evaluate
    wip_result = self.evaluate_into_dir(config)
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 470, in evaluate_into_dir
    return oscap_helpers.evaluate(self, config)
  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 300, in evaluate
    args = get_evaluation_args(spec, config)
  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 275, in get_evaluation_args
    ret.extend(spec.get_oscap_arguments(config))
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 444, in get_oscap_arguments
    ret.append(config.get_cve_feed(self.get_cpe_ids(config)))
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 347, in get_cpe_ids
    self.target, config
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 529, in detect_CPEs_of_target
    results, stdout, stderr, exit_code = es.evaluate(config)
  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 499, in evaluate
    (stdout, stderr, e)
RuntimeError: Failed to read results.xml of EvaluationSpec evaluation.
stdout:


stderr:
OpenSCAP Error: Probe at sd=32 (system_info) reported an error: Invalid type, value or format [oval_probe_ext.c:393]
Unable to receive a message from probe [oval_probe_ext.c:579]
Failed to create a new agent session. [oval_session.c:320]


exception: [Errno 2] No such file or directory: '/var/lib/oscapd/work_in_progress/x8zu0__d/results.xml'
INFO:[100.00%] Scanned target 'chroot:///scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3'
Unmounted /run/atomic/2017-03-02-14-24-44-627828/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3
u'/scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3'
Traceback (most recent call last):
  File "./atomic", line 189, in <module>
    sys.exit(_func())
  File "/root/atomic/Atomic/scan.py", line 166, in scan
    self.output_results()
  File "/root/atomic/Atomic/scan.py", line 274, in output_results
    uuid = self.mount_paths[json_results['UUID'].rstrip('/')] if len(self.args.rootfs) == 0 \
KeyError: u'/scanin/e4b79d4d89ab9b0aa873fd8dc99c652483bb645f317712ef30d5904ac2eafef3'
```

**Comment by miabbott** [comment]
Spoke with @baude about this yesterday.  He's already looking at it.

**Comment by baude** [comment]
given that the fedora openscap image has been deprecated, we good now ?

**Comment by miabbott** [comment]
The case of using the `registry.access.redhat.com/rhel7/openscap` scanner was fixed in #926 

And since the `docker.io/fedora/atomic_scan_openscap` scanner is no longer available, I think we are good here.
